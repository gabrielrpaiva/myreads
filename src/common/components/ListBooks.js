import React, { Component } from 'react';
import If from '../../If';
import RelatedBooks from '../../RelatedBooks'
import ShelfOptions from './ShelfOptions'


class ListBooks extends Component {


    render() {

        const { books, shelf, setUpdate, allShelfs, bookIdUpdate,
            filterRelatedBooks, classPopUp, relatedBooks, closePopUp, relatedBookId } = this.props

        let showingBooks
 
        if (shelf === "allShelfs") {

            showingBooks = books

        } else {

            if (typeof (books) !== "undefined" && books.length > 0) {

                if (shelf === "none") { showingBooks = books.filter((book) => typeof (book.shelf) === "undefined") }
                else { showingBooks = books.filter((book) => book.shelf === shelf) }

            }
            else { showingBooks = books }

        }

        if (typeof (showingBooks) === "undefined") {

            showingBooks = []

        }

        showingBooks.map(book =>
            typeof (book.shelf) === "undefined" ? book.shelf ="none" : book.shelf 
        ) 

        return (
            <ol className="books-grid">
                {showingBooks.map(book => (

                    <li key={book.id}>
                        <div className="book">
                            <div className="book-top">
                                <div className="book-cover" style={{ width: 128, height: 193, backgroundImage: `url("${book.imageLinks.thumbnail}")` }}></div>
                                <If test={book.shelf === 'read'}>
                                    <div onClick={(event) => filterRelatedBooks(book)}>
                                        <a className={relatedBookId === book.id  ? "book-shelf-read-loader" : "book-shelf-read"}></a>
                                    </div>
                                </If>

                                <div className={bookIdUpdate === book.id  ? "book-shelf-on-changer" : "book-shelf-changer"}>
                                    <ShelfOptions allShelfs={allShelfs} setUpdate={setUpdate} book={book} />
                                </div>
                            </div>
                            <div className="book-title">{book.title}</div>
                            <div className="book-authors">{book.authors}</div>
                            <div className="book-authors">{book.shelf}</div>
                            <If test={book.shelf === 'read'}>
                                <div id="popup1" className={`overlay${classPopUp}`}>
                                    <div className="popup">
                                        <a className="close" onClick={(event) => closePopUp()}>&times;</a>
                                        <RelatedBooks bookId={book.id} relatedBooks={relatedBooks}
                                            setUpdate={setUpdate}
                                            objShelfs={allShelfs}
                                            bookIdUpdate={bookIdUpdate} />
                                    </div>
                                </div>
                            </If>
                        </div>
                    </li>

                ))}
            </ol>
        )
    }
}

export default ListBooks