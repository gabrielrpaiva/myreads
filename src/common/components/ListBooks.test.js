import React from 'react';
import { shallow, mount } from "enzyme";
import ListBooks from './ListBooks';

/* it("renders without crashing", () => {
    expect(shallow(<ListBooks />)).toMatchSnapshot();
}); */



const allShelfs = {
    moveto: "Move to..",
    currentlyReading: "Currently Reading",
    wantToRead: "Want To Read",
    read: "Read",
    none: "None"
}


const setUpdate = jest.fn()

const bookIdUpdate = ""

const filterRelatedBooks = jest.fn()

const classPopUp = ""

const relatedBooks = []

const closePopUp = jest.fn()

const relatedBookId = ""

it("renders without crashing", () => {

    let books = [
        { id: 10, title: "Livro", authors: "Eu", shelf: "read", imageLinks: { thumbnail: "algo" } },
        { id: 263, title: "Livro Dois", authors: "Outro", shelf: "read", imageLinks: { thumbnail: "algo dois" } }
    ]

    let shelf = "allShelfs"
    expect(shallow(<ListBooks books={books}
        shelf={shelf}
        setUpdate={setUpdate}
        allShelfs={allShelfs}
        bookIdUpdate={bookIdUpdate}
        filterRelatedBooks={filterRelatedBooks}
        classPopUp={classPopUp}
        relatedBooks={relatedBooks}
        closePopUp={closePopUp}
        relatedBookId={relatedBookId} />)).toMatchSnapshot();
});


it("renders without crashing two", () => {


    let books = [
        { id: 10, title: "Livro", authors: "Eu", shelf: "read", imageLinks: { thumbnail: "algo" } },
        { id: 263, title: "Livro Dois", authors: "Outro", shelf: "read", imageLinks: { thumbnail: "algo dois" } }
    ]

    let shelf = "read"

    expect(shallow(<ListBooks books={books}
        shelf={shelf}
        setUpdate={setUpdate}
        allShelfs={allShelfs}
        bookIdUpdate={bookIdUpdate}
        filterRelatedBooks={filterRelatedBooks}
        classPopUp={classPopUp}
        relatedBooks={relatedBooks}
        closePopUp={closePopUp}
        relatedBookId={relatedBookId} />)).toMatchSnapshot();
});


it("renders without crashing three", () => {


    let books = [
        { id: 10, title: "Livro", authors: "Eu", shelf: "currentlyReading", imageLinks: { thumbnail: "algo" } },
        { id: 263, title: "Livro Dois", authors: "Outro", shelf: "currentlyReading", imageLinks: { thumbnail: "algo dois" } }
    ]

    let shelf = "none"

    expect(shallow(<ListBooks books={books}
        shelf={shelf}
        setUpdate={setUpdate}
        allShelfs={allShelfs}
        bookIdUpdate={bookIdUpdate}
        filterRelatedBooks={filterRelatedBooks}
        classPopUp={classPopUp}
        relatedBooks={relatedBooks}
        closePopUp={closePopUp}
        relatedBookId={relatedBookId} />)).toMatchSnapshot();
});



it("renders without crashing four", () => {

    let books = undefined

    let shelf = "none"

    expect(shallow(<ListBooks books={books}
        shelf={shelf}
        setUpdate={setUpdate}
        allShelfs={allShelfs}
        bookIdUpdate={bookIdUpdate}
        filterRelatedBooks={filterRelatedBooks}
        classPopUp={classPopUp}
        relatedBooks={relatedBooks}
        closePopUp={closePopUp}
        relatedBookId={relatedBookId} />)).toMatchSnapshot();
});

it("renders without crashing five", () => {


    let books = [
        { id: 10, title: "Livro", authors: "Eu", shelf: "read", imageLinks: { thumbnail: "algo" } },
        { id: 263, title: "Livro Dois", authors: "Outro", shelf: "read", imageLinks: { thumbnail: "algo dois" } }
    ]

    let shelf = "read"

    const test = mount(
        <ListBooks books={books}
            shelf={shelf}
            setUpdate={setUpdate}
            allShelfs={allShelfs}
            bookIdUpdate={bookIdUpdate}
            filterRelatedBooks={filterRelatedBooks}
            classPopUp={classPopUp}
            relatedBooks={relatedBooks}
            closePopUp={closePopUp}
            relatedBookId={relatedBookId} />
    );

     
    expect(test.find("a").length).toBe(4); 
});


