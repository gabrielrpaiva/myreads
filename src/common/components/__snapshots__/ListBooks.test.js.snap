// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<ol
  className="books-grid"
>
  <li>
    <div
      className="book"
    >
      <div
        className="book-top"
      >
        <div
          className="book-cover"
          style={
            Object {
              "backgroundImage": "url(\\"algo\\")",
              "height": 193,
              "width": 128,
            }
          }
        />
        <If
          test={true}
        >
          <div
            onClick={[Function]}
          >
            <a
              className="book-shelf-read"
            />
          </div>
        </If>
        <div
          className="book-shelf-changer"
        >
          <ShelfOptions
            book={
              Object {
                "authors": "Eu",
                "id": 10,
                "imageLinks": Object {
                  "thumbnail": "algo",
                },
                "shelf": "read",
                "title": "Livro",
              }
            }
            setUpdate={[Function]}
          />
        </div>
      </div>
      <div
        className="book-title"
      >
        Livro
      </div>
      <div
        className="book-authors"
      >
        Eu
      </div>
      <div
        className="book-authors"
      >
        read
      </div>
      <If
        test={true}
      >
        <div
          className="overlay"
          id="popup1"
        >
          <div
            className="popup"
          >
            <a
              className="close"
              onClick={[Function]}
            >
              ×
            </a>
            <RelatedBooks
              bookId={10}
              bookIdUpdate=""
              relatedBooks={Array []}
              setUpdate={[Function]}
            />
          </div>
        </div>
      </If>
    </div>
  </li>
  <li>
    <div
      className="book"
    >
      <div
        className="book-top"
      >
        <div
          className="book-cover"
          style={
            Object {
              "backgroundImage": "url(\\"algo dois\\")",
              "height": 193,
              "width": 128,
            }
          }
        />
        <If
          test={true}
        >
          <div
            onClick={[Function]}
          >
            <a
              className="book-shelf-read"
            />
          </div>
        </If>
        <div
          className="book-shelf-changer"
        >
          <ShelfOptions
            book={
              Object {
                "authors": "Outro",
                "id": 263,
                "imageLinks": Object {
                  "thumbnail": "algo dois",
                },
                "shelf": "read",
                "title": "Livro Dois",
              }
            }
            setUpdate={[Function]}
          />
        </div>
      </div>
      <div
        className="book-title"
      >
        Livro Dois
      </div>
      <div
        className="book-authors"
      >
        Outro
      </div>
      <div
        className="book-authors"
      >
        read
      </div>
      <If
        test={true}
      >
        <div
          className="overlay"
          id="popup1"
        >
          <div
            className="popup"
          >
            <a
              className="close"
              onClick={[Function]}
            >
              ×
            </a>
            <RelatedBooks
              bookId={263}
              bookIdUpdate=""
              relatedBooks={Array []}
              setUpdate={[Function]}
            />
          </div>
        </div>
      </If>
    </div>
  </li>
</ol>
`;

exports[`renders without crashing four 1`] = `
<ol
  className="books-grid"
/>
`;

exports[`renders without crashing three 1`] = `
<ol
  className="books-grid"
/>
`;

exports[`renders without crashing two 1`] = `
<ol
  className="books-grid"
>
  <li>
    <div
      className="book"
    >
      <div
        className="book-top"
      >
        <div
          className="book-cover"
          style={
            Object {
              "backgroundImage": "url(\\"algo\\")",
              "height": 193,
              "width": 128,
            }
          }
        />
        <If
          test={true}
        >
          <div
            onClick={[Function]}
          >
            <a
              className="book-shelf-read"
            />
          </div>
        </If>
        <div
          className="book-shelf-changer"
        >
          <ShelfOptions
            book={
              Object {
                "authors": "Eu",
                "id": 10,
                "imageLinks": Object {
                  "thumbnail": "algo",
                },
                "shelf": "read",
                "title": "Livro",
              }
            }
            setUpdate={[Function]}
          />
        </div>
      </div>
      <div
        className="book-title"
      >
        Livro
      </div>
      <div
        className="book-authors"
      >
        Eu
      </div>
      <div
        className="book-authors"
      >
        read
      </div>
      <If
        test={true}
      >
        <div
          className="overlay"
          id="popup1"
        >
          <div
            className="popup"
          >
            <a
              className="close"
              onClick={[Function]}
            >
              ×
            </a>
            <RelatedBooks
              bookId={10}
              bookIdUpdate=""
              relatedBooks={Array []}
              setUpdate={[Function]}
            />
          </div>
        </div>
      </If>
    </div>
  </li>
  <li>
    <div
      className="book"
    >
      <div
        className="book-top"
      >
        <div
          className="book-cover"
          style={
            Object {
              "backgroundImage": "url(\\"algo dois\\")",
              "height": 193,
              "width": 128,
            }
          }
        />
        <If
          test={true}
        >
          <div
            onClick={[Function]}
          >
            <a
              className="book-shelf-read"
            />
          </div>
        </If>
        <div
          className="book-shelf-changer"
        >
          <ShelfOptions
            book={
              Object {
                "authors": "Outro",
                "id": 263,
                "imageLinks": Object {
                  "thumbnail": "algo dois",
                },
                "shelf": "read",
                "title": "Livro Dois",
              }
            }
            setUpdate={[Function]}
          />
        </div>
      </div>
      <div
        className="book-title"
      >
        Livro Dois
      </div>
      <div
        className="book-authors"
      >
        Outro
      </div>
      <div
        className="book-authors"
      >
        read
      </div>
      <If
        test={true}
      >
        <div
          className="overlay"
          id="popup1"
        >
          <div
            className="popup"
          >
            <a
              className="close"
              onClick={[Function]}
            >
              ×
            </a>
            <RelatedBooks
              bookId={263}
              bookIdUpdate=""
              relatedBooks={Array []}
              setUpdate={[Function]}
            />
          </div>
        </div>
      </If>
    </div>
  </li>
</ol>
`;
